crypto/spiffe/spiffe.go

```diff
@@ -175,27 +202,50 @@ func (s *SPIFFE) fetchIdentityCertificate(ctx context.Context) (*x509svid.SVID,
 		return nil, fmt.Errorf("failed to create sidecar csr: %w", err)
 	}
 
-	workloadcert, err := s.requestSVIDFn(ctx, csrDER)
+	svidResponse, err := s.requestSVIDFn(ctx, csrDER)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(workloadcert) == 0 {
+	if len(svidResponse.X509Certificates) == 0 {
 		return nil, errors.New("no certificates received from sentry")
 	}
 
-	spiffeID, err := x509svid.IDFromCert(workloadcert[0])
+	spiffeID, err := x509svid.IDFromCert(svidResponse.X509Certificates[0])
 	if err != nil {
 		return nil, fmt.Errorf("error parsing spiffe id from newly signed certificate: %w", err)
 	}
 
+	identity := &Identity{
+		X509SVID: &x509svid.SVID{
+			ID:           spiffeID,
+			Certificates: svidResponse.X509Certificates,
+			PrivateKey:   key,
+		},
+	}
+
+	// If we have a JWT token, parse it and include it in the identity
+	if svidResponse.JWT != "" {
+		// we are using ParseInsecure here as the expectation is that the
+		// requestSVIDFn will have already parsed and validate the JWT SVID
+		// before returning it.
```

Comment:
See `newRequestFn` in https://github.com/dapr/dapr/pull/8662/files for token validation